File: python/priority-queue.py
class PriorityQueue:
    '''
    Creating A Priority Queue.

    Functions
    ---------
    enqueue: add element to queue.
    dequeue: remove element from queue.
    isempty: checks whether queue is empty.
    print: prints itself.
    '''
    def __init__(self,*elements):
        self.list = list(elements)

    def enqueue(self,element,priority):
        if len(self.list) == 0:
            self.list.append([priority,element])
        else:
            for i in range(len(self.list)):
                if priority == self.list[i][0]:
                    self.list[i].append(element)
                    break
            else:
                self.list.append([priority,element])

    def dequeue(self):
        for i in self.list:
            if len(i) != 1:
                return i.pop(1)
    
    def isempty(self):
        return not bool(len(self.list))

    def print(self):
        self.list.sort()
        print(self.list)

if __name__ == "__main__":
    q = PriorityQueue()
    q.enqueue("1", 6)
    q.print()
    q.enqueue("2", 1)
    q.print()
    q.enqueue("3", 2)
    q.print()
    q.enqueue("4", 2)
    q.print()
    q.dequeue()
    q.print()
    q.dequeue()
    q.print()

File: javascript/priorityqueue.js
class PriorityQueue{
	constructor(){
		this.array = []
	}
	enqueue(ele,priority){
		if(this.array.length == 0){
			this.array.push([priority,ele])
		}
		else{
			let isadded = false
			for(let i = 0;i<this.array.length;i++){
				if(this.array[i][0] == priority){
					this.array[i].push(ele)
					isadded = true
				}
			}
			if(!isadded){
				this.array.push([priority,ele])
			}
		}
	}
	dequeue(){
		for(let i = 0;i<this.array.length;i++){
			if(this.array[i].length != 1){
				return this.array[i].splice(1,1)
				break
			}
		}
	}
	print(){
		console.log(this.array)
	}
}

// var pq = new PriorityQueue()
// pq.enqueue('10',1)
// pq.print()
// pq.enqueue('198',3)
// pq.print()
// pq.enqueue('2',5)
// pq.enqueue('53',10)
// pq.print()
// pq.dequeue()
// pq.print()
// pq.dequeue()
// pq.print()